src/cs50/__init__.py

```diff
@@ -1,34 +1,48 @@
+import os
 import sys
 
-from .cs50 import eprint, get_char, get_float, get_int, get_string
 try:
-    from .cs50 import get_long
-except:
-    pass
 
-from . import flask
+    # Save student's sys.path
+    path = sys.path[:]
 
+    # In case student has files that shadow packages
+    sys.path = [p for p in sys.path if p not in ("", os.getcwd())]
```

Comment:
Per the docs, the directory in which the script is run (or `""` if in the interpreter) is always `sys.path[0]`. With this in mind, I think we could just do `del sys.path[0]` for the same effect (and thus avoid having to reconstruct the whole list). 

Alternatively we could do e.g.
```
try:
    cwd = sys.path.pop(0)
   # ...
finally:
    sys.path.insert(0, cwd)
```
which avoids having to store two copies in the first place.

Reply:
Hm, technically, a student could have prepended a path of their own? Feels more robust not to make any assumptions? And cost of reconstruction is pretty trivial.